language/compiler/ir-to-bytecode/src/compiler.rs

```diff
@@ -563,15 +579,27 @@ pub fn compile_module<'a, T: 'a + ModuleAccess>(
 }
 
 fn compile_explicit_dependency_declarations(
-    context: &mut Context,
+    outer_context: &mut Context,
+    address_opt: Option<AccountAddress>,
+    imports: Vec<ImportDefinition>,
     dependencies: Vec<ModuleDependency>,
-) -> Result<()> {
+) -> Result<Vec<CompiledModule>> {
+    let outer_depedencies = outer_context.dependencies().clone();
+    let mut compiled_deps = vec![];
     for dependency in dependencies {
         let ModuleDependency {
             name: mname,
             structs,
             functions,
         } = dependency;
+        let current_module = outer_context.module_ident(&mname)?;
+        let mut context = Context::new(outer_depedencies.clone(), Some(current_module.clone()))?;
+        // TODO find a way to accumulate depedencies. Borrow issues galore on attempts
```

Comment:
Did you intend to fix this before landing?

Reply:
Not really :P 

I think I might have an idea (gained from working on another PR) but it is a really tricky thing to fix

Reply:
@zekun000 showed me the rental crate which let me solve this issue. I added another commit to this PR with the fix